app/models/project.rb

```diff
@@ -293,6 +295,18 @@ def open_for_contributions?
     pluck_from_database(:open_for_contributions)
   end
 
+  def all_tags=(names)
+    self.tags = names.split(',').map do |name|
+      Tag.find_or_create_by(name: name.strip) do |tag|
```

Comment:
The idea is to save the exact user input and a slug to help searching later, if we look for slug first there is no way to recover the information later, that is, there is no way of going from slug to name.